src/prefect/runner.py

```diff
@@ -623,6 +629,31 @@ async def _get_scheduled_flow_runs(
         self._logger.debug(f"Discovered {len(scheduled_flow_runs)} scheduled_flow_runs")
         return scheduled_flow_runs
 
+    def _acquire_limit_slot(self, flow_run_id: str) -> bool:
+        """
+        Enforces flow run limit set on runner.
+
+        Returns:
+            - bool: True if a slot was acquired, False otherwise.
+        """
+        try:
+            if self._limiter:
+                self._limiter.acquire_on_behalf_of_nowait(flow_run_id)
+            return True
+        except anyio.WouldBlock:
+            self._logger.info(
+                f"Flow run limit reached; {self._limiter.borrowed_tokens} flow runs"
+                " in progress."
+            )
+            return False
+
+    def _release_limit_slot(self, flow_run_id: str) -> None:
+        """
+        Frees up a slot taken by the given flow run id.
+        """
+        if self._limiter:
+            self._limiter.release_on_behalf_of(flow_run_id)
+
```

Comment:
Would it be a good idea to ensure this behavior is working correctly?

Reply:
yea i can add a test or two

Reply:
updated with a test 👍

Reply:
nice!